### ISSUE TITLE
C99 build failure on FreeBSD

### ISSUE THREAD

**Issue by dch** [h2o/h2o#2933]
```
$ cd (mktemp -dt h2o) && cmake -G Ninja /repos/h2o && ninja

...

[269/437] Building C object CMakeFiles/h2o.dir/deps/libgkc/gkc.c.o
/repos/h2o/deps/libgkc/gkc.c:338:12: warning: variable 'gi' set but not used [-Wunused-but-set-variable]
    double gi = 0;
           ^
1 warning generated.
[420/437] Building C object CMakeFiles/h2o.dir/src/main.c.o
FAILED: CMakeFiles/h2o.dir/src/main.c.o
ccache /usr/bin/cc -DH2O_HAS_GITREV_H -DLIBC_HAS_BACKTRACE -I/repos/h2o/deps/mruby/include -I/repos/h2o/deps/mruby-input-stream/src -I/repos/h2o/include -I/repos/h2o/deps/cloexec -I/repos/h2o/deps/brotli/c/include -I/repos/h2o/deps/golombset -I/repos/h2o/deps/hiredis -I/repos/h2o/deps/libgkc -I/repos/h2o/deps/libyrmcds -I/repos/h2o/deps/klib -I/repos/h2o/deps/neverbleed -I/repos/h2o/deps/picohttpparser -I/repos/h2o/deps/picotest -I/repos/h2o/deps/picotls/deps/cifra/src/ext -I/repos/h2o/deps/picotls/deps/cifra/src -I/repos/h2o/deps/picotls/deps/micro-ecc-I/repos/h2o/deps/picotls/include -I/repos/h2o/deps/quicly/include -I/repos/h2o/deps/yaml/include -I/repos/h2o/deps/yoml -I/tmp/h2o -I/usr/local/include -std=c99 -g3 -Wall -Wno-unused-value -Wno-unused-function -Wno-nullability-completeness -Wno-expansion-to-defined -Werror=implicit-function-declaration -Werror=incompatible-pointer-types  -pthread -DH2O_ROOT="/usr/local" -DH2O_CONFIG_PATH="/usr/local/etc/h2o.conf" -pthread -DH2O_HAS_PTHREAD_SETAFFINITY_NP -O2 -DH2O_USE_LIBUV=0 -DH2O_USE_BROTLI=1 -DQUICLY_USE_TRACER=1 -DH2O_USE_MRUBY=1 -MD -MT CMakeFiles/h2o.dir/src/main.c.o -MF CMakeFiles/h2o.dir/src/main.c.o.d -o CMakeFiles/h2o.dir/src/main.c.o -c /repos/h2o/src/main.c
/repos/h2o/src/main.c:3048:13: error: implicit declaration of function 'pthread_setaffinity_np' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        if (pthread_setaffinity_np(pthread_self(), sizeof(cpu_set), &cpu_set) != 0)
            ^
1 error generated.
[429/437] Building C object CMakeFiles/h2o.dir/deps/brotli/c/enc/backward_references.c.
```

**Comment by dch** [comment]
In principle, h2o's cmakelists.txt should accommodate this, but clearly doesn't. See #2944 for a possible solution.

**Comment by kazuho** [comment]
Addressed by #2942.
